packages/gatsby/src/utils/gatsby-webpack-virtual-modules.ts

```diff
@@ -32,10 +32,12 @@ export class GatsbyWebpackVirtualModules {
   }
 }
 
+export function getAbsolutePathForVirtualModule(filePath: string): string {
+  return path.join(process.cwd(), `_this_is_virtual_fs_path_`, filePath)
```

Comment:
The `_this_is_virtual_fs_path_` here is mostly to help with any future debugging (makes it easier to grep those keywords), it's not needed to make it work (tho it's still good to group all of those under unlikely name, because modules - be it fs or virtual share same path namespace so this also should prevent collisions - unless someone names their directory like that)